readme.md

```diff
@@ -190,6 +190,17 @@ The evaluation data context is used to provide the expression with variable refe
 To reference the nested reference, please use "." delimiter, e.g.:
 ```$address.city```
 
+You can also use square brackets to access a specific element of an array:
```

Comment:
Could we slightly reformulate here, if it make sense:

**Accessing Array Element:**
```$options[1]```

**Accessing Array Element via Reference:**
```$options[{index}]```
- **index** reference is resolved within the data context as an array index.

**Nested Referencing**
```$address.{segment}```
- **segment** reference is resolved within the data context as a property key.

**Composite Reference Key**
```$shape{shapeType}```
- **shapeType** reference is resolved within the data context, and inserted into the outer reference key.
- E.g. **shapeType** resolved as "**B**" would compose **$shapeB** outer reference.
- This resolution could be n-nested.

```
// Data context
const ctx = {
  name: 'peter',
  country: 'canada',
  age: 21,
  options: [1,2,3],
  address: {
    city: 'Toronto',
    country: 'Canada',
  },
  index: 2,
  segment: 'city',
  shapeA: 'box,
  shapeB: 'circle',
  shapeType: 'B'
}
```

Reply:
13b35e60462b79c6125c239e32eb7acd8406cdba